sapi/fpm/fpm/fpm_stdio.c

```diff
@@ -144,12 +144,24 @@ static void fpm_stdio_child_said(struct fpm_event_s *ev, short which, void *arg)
 		zlog_stream_init_ex(log_stream, ZLOG_WARNING, STDERR_FILENO);
 		zlog_stream_set_decorating(log_stream, child->wp->config->decorate_workers_output);
 		zlog_stream_set_wrapping(log_stream, ZLOG_TRUE);
-		zlog_stream_set_msg_prefix(log_stream, "[pool %s] child %d said into %s: ",
-				child->wp->config->name, (int) child->pid, is_stdout ? "stdout" : "stderr");
 		zlog_stream_set_msg_quoting(log_stream, ZLOG_TRUE);
 	} else {
 		log_stream = child->log_stream;
+		// if fd type (stdout/stderr) is changed, then the stream will be finished
+		if (log_stream->is_stdout != (unsigned int)is_stdout) {
+			zlog_stream_finish(log_stream);
+		}
+		// free stream's msg_prefix for reset
+		if (log_stream->msg_prefix) {
```

Comment:
This would be better to handle inside `zlog_stream_set_msg_prefix` as it won't be necessary to do another allocation (`"stdout"` and `"stderr`" have the same length). However to make it generic there should be a check added. Basically something like:
```
zlog_bool zlog_stream_set_msg_prefix(struct zlog_stream *stream, const char *fmt, ...) /* {{{ */
{
	char buf[MAX_WRAPPING_PREFIX_LENGTH];
	size_t len;
	va_list args;

	if (!stream->decorate) {
		return ZLOG_TRUE;
	}

	va_start(args, fmt);
	len = vsnprintf(buf, MAX_WRAPPING_PREFIX_LENGTH - 1, fmt, args);
	va_end(args);

	if (stream->msg_prefix_len < len) {
		stream->msg_prefix = stream->msg_prefix_len ? realloc(stream->msg_prefix, len + 1) : malloc(len + 1);
		if (stream->msg_prefix == NULL) {
			return ZLOG_FALSE;
		}
	}
	memcpy(stream->msg_prefix, buf, len);
	stream->msg_prefix[len] = 0;
	stream->msg_prefix_len = len;

	return len;
}
```